internals-js/src/specs/coreSpec.ts

```diff
@@ -552,6 +552,26 @@ export class CoreSpecDefinition extends FeatureDefinition {
     return feature.addElementsToSchema(schema);
   }
 
+  applyFeatureAsLink(schema: Schema, feature: FeatureDefinition, purpose?: CorePurpose, imports?: CoreImport[]): GraphQLError[] {
+    const existing = schema.schemaDefinition.appliedDirectivesOf("link").find((link) => link.arguments().url === feature.toString());
+    if (existing) {
+      existing.remove();
+    }
+
+    const coreDirective = this.coreDirective(schema);
+    const args = {
+      url: feature.toString(),
```

Comment:
Nit, but I prefer naming the type rather than casting. That way you aren't fighting the type system.

```
    const args: LinkDirectiveArgs = {
      url: feature.toString(),
      import: (existing?.arguments().import ?? []).concat(imports?.map((i) => i.as ? { name: `@${i.name}`, as: `@${i.as}` } : `@${i.name}`)),
      feature: undefined,
    };
```